core/services/workflows/workflow.go

```diff
@@ -97,20 +100,170 @@ type stepDefinition struct {
 	Config map[string]any `json:"config" jsonschema:"required"`
 }
 
-// workflowSpec is the parsed representation of a workflow.
-type workflowSpec struct {
+// toStepDefinition converts a stepDefinitionYaml to a stepDefinition.
+//
+// `stepDefinition` is the converged representation of a step in a workflow.
+func (s stepDefinitionYaml) toStepDefinition() stepDefinition {
```

Comment:
@HenryNguyen5 nit: Can we move this next to the definition of stepDefinitionYaml?

Reply:
Ugh sorry !